File: 06Adapter/Students/2019/PawelskiBartlomiej/twoWayAdapter/Seabird.java
package twoWayAdapter;

public class Seabird extends Seacraft implements IAircraft {
    int height = 0;

    @Override
    public boolean airbone() {
        return height > 50;
    }

    @Override
    public void takeOff() {
        while (!airbone()) increaseRevs();
    }

    @Override
    public int height() {
        return height;
    }

    @Override
    public void increaseRevs() {
        super.increaseRevs();
        height += speed() > 40 ? 100 : 0;
    }
}

File: 06Adapter/Students/2016/mbowzyk/src/main/kotlin/twoWayAdapter/SeaBird.kt
package adapter.twoWayAdapter

class SeaBird : SeaCraft(), AirCraftInterface {

    override var airborn: Boolean = false
    override var height: Int = 0

    override fun takeOff() {
        while (!airborn)
            increaseRevs()
        println("it's flying!!")
    }

    override fun increaseRevs() {
        super.increaseRevs()

        if (speed > 40)
            height += 100
        airborn = (height > 50)
    }
}